### PR TITLE
Enable re-using polaris-service tests

### PR DESCRIPTION
**Pull request by andrew4699**
# Description

Generates a new build artifact with the test sources. This lets you re-use test utils and extend entire test classes. For example, if you have a project that includes Polaris, you can now write your own lightweight wrapper test to run the Polaris tests with your own config/environment.

When testing this end-to-end with a class that extends a Polaris test, I ran into an issue where PolarisConnectionExtension and the test classes decided on a different realm identifier, which meant that credentials were created in a different realm than the test ran in. This happened because the realm name was derived from the class names but the class name is tricky to fetch when using inheritance. I changed the test classes to just re-use the same realm name from PolarisConnectionExtension.

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tested the JAR is generated when running `gradle assemble`:
Screenshot 2024-09-12 at 10 42 57 AM

As a smoke test, validated that `gradle build` still works and the server still starts up.

# Checklist:

- [X] I have performed a self-review of my code
- [X] My changes generate no new warnings

### CODE REVIEW

**Review by dimas-b — COMMENTED** [review]
The PR LGTM overall... Just sharing some ideas about JUnit5 :)

File: polaris-service/src/test/java/org/apache/polaris/service/admin/PolarisOverlappingCatalogTest.java [hunk 1]
```diff
@@ -63,7 +63,7 @@ public class PolarisOverlappingCatalogTest {
   @BeforeAll
   public static void setup(PolarisConnectionExtension.PolarisToken adminToken) throws IOException {
     userToken = adminToken.token();
-    realm = PolarisConnectionExtension.getTestRealm(PolarisOverlappingCatalogTest.class);
+    realm = PolarisConnectionExtension.getTestRealm();
```
[the remark this answers is not in the document]
Comment (andrew4699) [reply polaris-service/src/test/java/org/apache/polaris/service/admin/PolarisOverlappingCatalogTest.java]: I didn't know that was possible, updated the PR to get rid of `getTestRealm()`!

File: polaris-service/src/test/java/org/apache/polaris/service/admin/PolarisOverlappingTableTest.java [hunk 2]
```diff
@@ -135,9 +136,9 @@ public String toString() {
   }
 
   @BeforeEach
-  public void setup(PolarisConnectionExtension.PolarisToken adminToken) {
+  public void setup(PolarisConnectionExtension.PolarisToken adminToken, @PolarisRealm String realm) {
     userToken = adminToken.token();
-    realm = PolarisConnectionExtension.getTestRealm(PolarisServiceImplIntegrationTest.class);
+    PolarisOverlappingTableTest.realm = realm;
```
Comment (dimas-b) [inline polaris-service/src/test/java/org/apache/polaris/service/admin/PolarisOverlappingTableTest.java:141]: nit: `real` can be non-static now

**Review by dimas-b — APPROVED** [review]
Aside from the minor comments above. LGTM :)

**Comment by andrew4699** [comment]
Just fixed the spotless checks and removed `static realm` in classes where there weren't static helpers that relied on it.

File: polaris-service/src/test/java/org/apache/polaris/service/admin/PolarisServiceImplIntegrationTest.java [hunk 3]
```diff
@@ -99,9 +100,11 @@ public class PolarisServiceImplIntegrationTest {
   private static String realm;
 
   @BeforeAll
-  public static void setup(PolarisConnectionExtension.PolarisToken adminToken) throws IOException {
+  public static void setup(
+      PolarisConnectionExtension.PolarisToken adminToken, @PolarisRealm String realm)
+      throws IOException {
     userToken = adminToken.token();
-    realm = PolarisConnectionExtension.getTestRealm(PolarisServiceImplIntegrationTest.class);
+    PolarisServiceImplIntegrationTest.realm = realm;
```
Comment (dimas-b) [inline polaris-service/src/test/java/org/apache/polaris/service/admin/PolarisServiceImplIntegrationTest.java:107]: nit: I'd personally remove the field qualifier here as it was not present before.

Comment (andrew4699) [reply polaris-service/src/test/java/org/apache/polaris/service/admin/PolarisServiceImplIntegrationTest.java]: Updated to remove it in a couple of places.

Comment (dimas-b) [reply polaris-service/src/test/java/org/apache/polaris/service/admin/PolarisServiceImplIntegrationTest.java]: thx :+1:

**Review by dimas-b — APPROVED** [review]

**Comment by flyrain** [comment]
Thanks @andrew4699 for the PR. Thanks @dimas-b for the fix.

**Pull request merged**